Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -1635,3 +1619,61 @@ void bsg_kscrashreport_logCrash(const BSG_KSCrash_Context *const crashContext) {
     bsg_kscrw_i_logCrashType(crash);
     bsg_kscrw_i_logCrashThreadBacktrace(&crashContext->crash);
 }
+
+static size_t bsg_g_thread_json_size = sizeof(char) * 32 * 1024;
+static char *bsg_g_thread_json_data;
+
+int bsg_kscrw_i_collectJsonData(const char *const data, const size_t length, void *const userData) {
+    size_t new_size = strlen(bsg_g_thread_json_data) + length;
+    char terminated_str[length + 1];
+    memcpy(terminated_str, data, length);
+    terminated_str[length] = '\0';
+
+    if (new_size > bsg_g_thread_json_size) {
```

Comment:
I believe the data may or may not have a terminator already (I can see calls from KS code that passes data as a string literal which will be terminated.

Therefore by adding our own and then `strlen`-ing the array, we will be able to pass it to `strlen` to give us the length we need. So I think you can do away with `new_size` and just use:

```suggestion
    if (strlen(bsg_g_thread_json_data) + strlen(terminated_str) > bsg_g_thread_json_size) {
```

Reply:
Updated code to use this suggestion.